amqp/readers_writers.go

```diff
@@ -92,19 +92,27 @@ func WriteFrame(wr io.Writer, frame *Frame) (err error) {
 	}
 
 	return nil
-
-	//return binary.Write(wr, binary.BigEndian, frameBuffer.Bytes())
 }
 
 // ReadOctet reads octet (byte)
 func ReadOctet(r io.Reader) (data byte, err error) {
-	err = binary.Read(r, binary.BigEndian, &data)
+	b := make([]byte, 1)
```

Comment:
`make` will likely allocate on the heap, which will be expensive for each byte read.
Allocating the slice on the stack may be a better idea,

``` go
	var b [1]byte

	if _, err = io.ReadFull(r, b[:]); err != nil {
		return
	}
	data = b[0]
```

Reply:
Looks like nice idea, fixed